python/ml4ir/base/pipeline.py

```diff
@@ -231,24 +231,6 @@ def validate_args(self):
 
         return self
 
-    def finish(self):
```

Comment:
this version of `finish` is not used anywhere, correct me if I a wrong. Lower in the same file there is another definition of it. Removing it to clean code.

Reply:
Yeah. I think this is an artifact of rebase and merge. I'm fixing it in my follow up PR. But you can do it here as well.